functorch/_src/vmap.py

```diff
@@ -203,7 +203,7 @@ def _get_name(func: Callable):
 # on BatchedTensors perform the batched operations that the user is asking for.
 
 
-def vmap(func: Callable, in_dims: in_dims_t = 0, out_dims: out_dims_t = 0) -> Callable:
+def vmap(func: Callable, in_dims: in_dims_t = 0, out_dims: out_dims_t = 0, use_batched_random=True) -> Callable:
```

Comment:
We should bikeshed this name...
I called it `use_batched_randomness` to mean that if True, each batch from vmap will have different random numbers. But I can definitely see a user thinking this meant that the randomness will be the same across batches (as I myself got confused while writing this sometimes)

Some potential jumping off points:
  - `use_batched_random`
  - `use_different_random`
  - `matching_random_across_batches`